website/docs/tutorials/working-with-files.md

```diff
@@ -4,8 +4,127 @@ title: Working with Files
+the fetched file as a blob. It is then up to you to decode it appropriately.
+
+```typescript
+import { unstable_fetchFile } from "lit-solid";
+
+const response = await unstable_fetchFile(
+  "https://example.com/some/interesting/file"
+);
+if (response.ok) {
+  const myFile = await response.blob();
+}
+```
+
+## Deleting a file
+
+Deleting a file is also a simple operation: you just erase the content available at a certain URL.
+
+```typescript
+import { unstable_fetchFile } from "lit-solid";
+
+const response = await unstable_deleteFile(
+  "https://example.com/some/boring/file"
+);
+if (response.ok) {
+  console.log("File deleted !");
+}
+```
+
+## Writing a file
+
+There are two approaches to writing files:
+
+1. you know exactly at which IRI your file should be saved (potentially overwriting any data that sat there previously)
+2. you know what resource should be the parent of your file, like saving it into a folder.
+
+### Writing a file directly at a URL
+
+With this approach, if the request succeeds, you know exactly what is the URL of your file.
+
+```typescript
+import { unstable_overwriteFile } from "lit-solid";
+
+const response = await unstable_overwriteFile(
+  "https://example.com/some/new/file",
+  new Blob(["This is a plain piece of text"], { type: "plain/text" })
+);
+if (response.ok) {
+  console.log("File saved !");
+}
+```
+
+### Saving a file into a parent resource
+
+With this approach, you kindly ask the server to come up with a name for your file, potentially using a `slug` if you
+provide one. Note that there is no guarantee about how the server will use the slug, if at all. In any case, if the slug
+you provide matches a file that already exists under the same target resource, the server will create a new name for your
+file so that no content is overwritten.
+
+This means that you don't control the final name of your file though. To keep track of the name the server gave to your
+file, you'll have to look up the `Location` header in the response, as shows the code snippet below:
+
+```typescript
+import { unstable_saveFileInContainer } from "lit-solid";
+
+const response = await unstable_saveFileInContainer(
+  "https://example.com/some/folder",
+  new Blob(["This is a plain piece of text"], { type: "plain/text" }),
+  { slug: "new-file" }
+);
+if (response.ok) {
+  const headers = await response.headers;
+  console.log(`File saved at ${headers.get("Location")}`);
+}
+```
+
+Note that the returned `Location` should be relative to the server's origin, so in the previous example `Location` might
+be `/some/folder/new-file-3869a250`, which means the file is saved at the URL `https://example.com/some/folder/new-file-3869a250`.
+
+## Customizing the requests
+
+If you need to customize the request eventually sent to the server, you can do so by using the `init` optionnal parameter.
+`init` is conform to the `RequestInit` type of the `fetch` API. For instance, the following code snippet sets a custom
```

Comment:
```suggestion
`init` conforms to the [`init` parameter](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters) of the `fetch` API. For instance, the following code snippet sets a custom
```

(`RequestInit` is the name the authors of the TypeScript type definitions gave it, but is not a universally known term, I think.)